src/jsifier.js

```diff
@@ -282,6 +282,12 @@ function JSify(data, functionsOnly) {
       if (typeof snippet === 'string') {
         var target = LibraryManager.library[snippet];
         if (target) {
+          if (LibraryManager.library[snippet + '__asm'] && !LibraryManager.library[ident + '__asm']) {
+            error('A non-asm.js/wasm function "' + snippet + '" cannot be constructed as an alias of an asm.js/wasm function "' + ident + '"! (create a helper function to call into asm.js if you want to keep the alias)');
+          }
+          if (!LibraryManager.library[snippet + '__asm'] && LibraryManager.library[ident + '__asm']) {
+            error('An asm.js/wasm function "' + snippet + '" cannot be constructed as an alias of a non-asm.js/wasm function "' + ident + '"! (the JS function needs to be imported to asm.js/wasm)');
+          }
```

Comment:
Why is this necessary?

Is it still necessary after we get closure and metadce fixed?

Reply:
It is wrong to have a `library_foo.js` file with

```javascript
  foo__asm: 'true',
  foo: function() {
    (asm code here)
  },

  foo_alias: 'foo',
```

because that would generate code `var foo_alias = foo;` way before the asm/wasm module has even been loaded, so `foo_alias` ends up being `undefined`, as `foo` is received as an export only later. The above change enforces that one does not alias across non-asmjs and asmjs functions.